Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -2294,101 +2299,33 @@ static void appendPipelineAppSinkCapsChanged(GObject*, GParamSpec*, AppendPipeli
 
 static GstPadProbeReturn appendPipelineAppsrcDataLeaving(GstPad*, GstPadProbeInfo* info, AppendPipeline* appendPipeline)
 {
-    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER) {
-        GstBuffer* buffer = GST_PAD_PROBE_INFO_BUFFER(info);
-        gsize bufferSize = gst_buffer_get_size(buffer);
+    ASSERT(GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_BUFFER);
 
-        TRACE_MEDIA_MESSAGE("buffer of size %" G_GSIZE_FORMAT " going thru", bufferSize);
+    GstBuffer* buffer = GST_PAD_PROBE_INFO_BUFFER(info);
+    gsize bufferSize = gst_buffer_get_size(buffer);
 
-        if (bufferSize > 0)
-            appendPipeline->reportEndOfAppendDataMarkNeeded();
+    TRACE_MEDIA_MESSAGE("buffer of size %" G_GSIZE_FORMAT " going thru", bufferSize);
 
-        return GST_PAD_PROBE_OK;
-    }
-
-#ifdef DEBUG_APPEND_PIPELINE_PADS
-    if (GST_PAD_PROBE_INFO_TYPE(info) & GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM) {
-        GstEvent* event = GST_PAD_PROBE_INFO_EVENT(info);
-        if (GST_EVENT_TYPE(event) != GST_EVENT_CUSTOM_DOWNSTREAM)
-            return GST_PAD_PROBE_OK;
-
-        const GstStructure* structure = gst_event_get_structure(event);
-        if (!gst_structure_has_name(structure, "end-of-append-data-mark"))
-            return GST_PAD_PROBE_OK;
-
-        guint id = gst_event_get_seqnum(event);
-        TRACE_MEDIA_MESSAGE("custom downstream event id=%u", id);
-
-        return GST_PAD_PROBE_OK;
-    }
-#endif
+    appendPipeline->markAtLeastABufferLeftAppsrc();
```

Comment:
Also return GST_PAD_PROBE_REMOVE; instead of manually calling gst_pad_probe_remove()